libs/ibm/langchain_ibm/chat_models.py

```diff
@@ -526,6 +579,30 @@ def lc_secrets(self) -> Dict[str, str]:
     @model_validator(mode="after")
     def validate_environment(self) -> Self:
         """Validate that credentials and python package exists in environment."""
+        self.params = self.params or {}
+
+        if isinstance(self.params, BaseSchema):
+            self.params = self.params.to_dict()
+
+        check_duplicate_chat_params(self.params, self.__dict__)
+
+        self.params.update(
+            {
+                k: v
+                for k, v in {
+                    "frequency_penalty": self.frequency_penalty,
+                    "logprobs": self.logprobs,
+                    "top_logprobs": self.top_logprobs,
+                    "max_tokens": self.max_tokens,
+                    "n": self.n,
+                    "presence_penalty": self.presence_penalty,
+                    "temperature": self.temperature,
+                    "top_p": self.top_p,
+                }.items()
+                if v is not None
+            }
+        )
+
```

Comment:
Can we somehow reuse here `_update_params` function?

Reply:
Probably it can not be reused. Here we are updating the default parameters, but in `_update_params ` function (renamed to `_merge_params `) we are only merge it without adding it to default params.